src/main/java/org/truffleruby/options/CommandLineParser.java

```diff
@@ -376,7 +377,7 @@ private void processArgument() {
                             extendedOption = extendedOption.substring(0, equals);
                         }
 
-                        config.getOptions().put(extendedOption, value);
+                        config.getOptions().put(RubyLanguage.ID + "." + extendedOption, value);
```

Comment:
I think this should be `Main.LANGUAGE_ID` to avoid introducing the RubyLanguage dependency back.